internal/controller/dnsrecord_healthchecks.go

```
@@ -0,0 +1,243 @@
+		dnsRecord.Status.HealthCheck = &v1alpha1.HealthCheckStatus{
+			Conditions: []metav1.Condition{},
+			Probes:     []v1alpha1.HealthCheckStatusProbe{},
+		}
+	}
+
+	for _, c := range dnsRecord.Status.HealthCheck.Conditions {
+		if c.Type == "healthProbesSynced" {
+			probesCondition = c
+			probesCondition.ObservedGeneration = dnsRecord.Generation
+		}
+	}
+
+	dnsRecord.Status.HealthCheck.Probes = []v1alpha1.HealthCheckStatusProbe{}
+
+	for _, result := range results {
+		status := true
+		if result.Result == provider.HealthCheckFailed {
+			status = false
+			allSynced = metav1.ConditionFalse
+		}
+
+		dnsRecord.Status.HealthCheck.Probes = append(dnsRecord.Status.HealthCheck.Probes, v1alpha1.HealthCheckStatusProbe{
+			ID:        result.ID,
+			IPAddress: result.IPAddress,
+			Host:      result.Host,
+			Synced:    status,
+		})
+	}
+
+	if probesCondition.Status != allSynced {
+		probesCondition.Status = allSynced
+		probesCondition.LastTransitionTime = metav1.Now()
+		if allSynced == metav1.ConditionTrue {
+			probesCondition.Reason = "AllProbesSynced"
+		} else {
+			probesCondition.Reason = "UnsyncedProbes"
+			probesCondition.Message = "some probes have not yet successfully synced to the DNS Provider"
+		}
+	}
+	dnsRecord.Status.HealthCheck.Conditions = []metav1.Condition{}
+	if len(results) > 0 {
+		dnsRecord.Status.HealthCheck.Conditions = []metav1.Condition{probesCondition}
+	}
+	return nil
+}
+
+func (r *DNSRecordReconciler) reconcileHealthCheckDeletion(ctx context.Context, dnsRecord *v1alpha1.DNSRecord) ([]provider.HealthCheckResult, error) {
+	log.Log.Info("deleting health checks")
+	dnsProvider, err := r.getDNSProvider(ctx, dnsRecord)
+	if err != nil {
+		log.Log.Info("error getting dns provider", "err", err)
+		return nil, err
+	}
+	reconciler := dnsProvider.HealthCheckReconciler()
+	var results []provider.HealthCheckResult
+	for _, endpoint := range dnsRecord.Spec.Endpoints {
+		address, _ := provider.GetEndpointAddress(endpoint)
```

should we be ignoring the error here?